enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,432 @@
+since we're providing enough possibilities to customize Image Registry storage backend.
+
+#### Cluster Ingress Operator
+
+Within the ingress operator PlatformType affects on two things:
+1. Choosing EndpointPublishingStrategy, which is `HostNetworkStrategyType` for "None" and unknown PlatformType
+2. Creating dns provider on some platforms. This logic does not engaged for "None" or unknown PlatformType
+
+Speaking about `LoadBalancer` EndpointPublishingStrategy - Cluster Ingress Operator might be reconfigured for using it
+as day2 operation. Operator itself  creates `Service` objects with correct provider-specific annotations, actual handling
+of such objects happens in a provider-specific [cloud controller manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/).
+
+#### Cluster Network Operator
+
+TBD
+
+### Workflow Description
+
+Since proposed change intended to have no effect during initial implementation, no specific user interaction is expected.
+"External" platform type would be treated the same as "None" platform.
+
+### API Extensions
+
+A new optional constant of `PlatformStatus` type will be added to the "openshift/api".
+
+```go
+const (
+	...
+    // ExternalPlatformType represent generic infrastructure provider. Provider-specific components should be supplemented separately.
+    ExternalPlatformType PlatformType = "External"
+	...
+)
+```
+
+Additionally, respective external platform spec and status should be added to the infrastructure resource. 
+
+```go
+type CloudControllerManagerMode string
+
+const (
+    // Cloud Controller Manager is enabled and expected to be supplied.
+	// Signaling MCO to set `--cloud-provider=external` flag to the kubelets.
+    CloudControllerManagerEnabled CloudControllerManagerMode = "Enabled"
+    // No Cloud Controller Manager is expected to be supplied.
+    // Signaling MCO not to set `--cloud-provider` flag to the kubelets.
+    CloudControllerManagerDisabled CloudControllerManagerMode = "Disabled"
+)
+
+// ExternalPlatformSpec holds the desired state for the generic External infrastructure provider.
+type ExternalPlatformSpec struct{
+    // CloudControllerManager determines whether or not an external Cloud Controller Manager is expected to 
+    // be presented in the cluster.
+    // For engaging an external Cloud Controller Manager, certain flags are expected to be set to the kubelets.
+    // https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager
+    //
+    // When enabled, the respective operator (machine config operator) should set `--cloud-provider=external` flag to the kubelet.
+    // When omitted or disabled, no `cloud-provider` flag should be set.
+    // +kubebuilder:validation:Enum=Enabled;Disabled
+    // +optional
+    CloudControllerManager CloudControllerManagerMode `json:"cloudControllerManager,omitempty"`
```

Comment:
I really wonder if we should be making something more generic that turns on/off more things, having a single CCM flag means we can never add any additional configuration here, a struct may be better. I think we need to co-ordinate this effort with the PlatformOperators vision and capabilities API and work out exactly how the three will go together in the long term

Reply:
i think that makes some sense about adding a struct to allow additional options in the future.

on the point about syncing with the PlatformOperators vision, imo this is a strong reason for us to start migrating this enhancement to the openshift/enhancements repo as a PR. i'd like to start getting wider feedback and i feel like we will be missing a lot of conversation if we keep adding here.